src/main/java/com/github/manolo8/darkbot/core/manager/PetManager.java

```diff
@@ -767,5 +787,19 @@ public PetGear get() {
         public @Nullable Boolean enablePet() {
             return pet.isEnabled();
         }
+
+        @Override
+        public @Nullable LocatorPick getNpcLocatorPick(@UnmodifiableView Collection<? extends LocatorPick> picks) {
```

Comment:
Instead of providing a collection and expecting each impl to filter thru it, would it make more sense to provide this in a similar fashion to `compareTo` at the API level?

Maybe even have both methods, and have defaults (pseudo-code-kinda):
```java
default LocatorPick getNpcLocatorPick(Collection<LocatorPick> ...) {
  LocatorPick pick = picks.stream().max(Comparator.nullsLast().comparingInt(this::getPriority)).orElse(null);
  return pick != null && getPriority(pick) != null ?  pick : null;
}

default Integer getPriority(LocatorPick pick) {
  NpcInfo npc = pick.getNpcInfo();
  return npc != null && npc.kill() && npc.has(PET_LOCATOR) ? npc.priority (): null;
}
```